Source/WebCore/platform/graphics/haiku/ShareableBitmapHaiku.cpp

```diff
@@ -71,14 +70,49 @@ void ShareableBitmap::paint(GraphicsContext& context, float scaleFactor, const I
 
 WebCore::PlatformImagePtr ShareableBitmap::createPlatformImage(WebCore::BackingStoreCopy, WebCore::ShouldInterpolate)
 {
-    // FIXME: This doesn't work with UNIX's implementation of shared memory atm.
-    // Or should we use Haiku's implementation of shared memory instead?
-    // NOTE: getBitmapUniqueID depends on bitmaps always being at offset 0 in
-    // their area.
-    // return WebCore::PlatformImagePtr(new BitmapRef(m_sharedMemory->area(), 0, bounds(), 0, /*m_configuration.platformColorSpace()*/ B_RGBA32, bytesPerRow()));
-
-    debugger("ShareableBitmap::createPlatformImage is unimplemented");
-    return nullptr;
+    // Creates a BBitmap (actually BitmapRef) that reads image data from the
+    // address given by data(). This address is in shared memory. The idea is
+    // that multiple processes can point to the same underlying bitmap data.
+    // One can draw, and the other can display.
+    //
+    // TODO: We create a new BBitmap every time this is called! AFAIK, BBitmap
```

Comment:
this shouldn't be a problem, app_server should be able to track that and notice that it already has an instance of the bitmap.

It is also possible to avoid most of this cost if you don't set B_BITMAP_ACCEPT_VIEWS (a bitmap that can't be drawn onto) or set B_NO_SERVER_LINK (to create a bitmap that you will never share with app_server for drawing). B_BITMAP_ACEPT_VIEWS is interesting because a bitmap that accepts views is actually paired with an offscreen BWindow (including a BLooper/BHandler). Unfortunately, it's not always possible to know in WebKit if a bitmap will be drawn onto in this way or not, which is why at the moment, the flag is always set. Maybe the receiving end of a shared bitmap can get away with not setting the flag, since it will not modify the bitmap or draw on it, but only consume it.

Also, is this really called several time with the same data? I expect that the shared bitmap will be created once, and then, one process draws into it and the other pushes it onto the screen, and after that it can be "recycled" (drawn onto again) for the next animation frame? Or is it new bitmaps everytime?

In the later case, we may need some other mechanism, maybe media kit buffers, rather than sharing BBitmap.

(this is all for later, first, let's get the basic thing to work, even slowly)